cgd_tx_eff/src/edu/ohsu/compbio/txeff/tx_eff_vcf.py

```diff
@@ -0,0 +1,296 @@
+'''
+Created on Apr 29, 2022
+
+@author: pleyte
+
+Update a VCF with the variant-transcript details.  
+'''
+
+import argparse
+import logging
+import sys
+import vcfpy
+from collections import defaultdict
+from enum import Enum
+from edu.ohsu.compbio.txeff.variant import Variant
+
+from edu.ohsu.compbio.txeff.util.tx_eff_csv import TxEffCsv
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+
+stream_handler = logging.StreamHandler()
+logging_format = '%(levelname)s: [%(filename)s:%(lineno)s - %(funcName)s()]: %(message)s'
+
+stream_format = logging.Formatter(logging_format)
+stream_handler.setFormatter(stream_format)
+stream_handler.setLevel(logging.DEBUG)
+logger.addHandler(stream_handler)
+
+VERSION = '0.0.1'
+
+class TranscriptEffect(Enum):
+    '''
+    The transcript effects that are added to VCF file
+    ''' 
+    TFX_GENE = 'TFX_GENE'
+    TFX_TRANSCRIPT = 'TFX_TRANSCRIPT'
+    TFX_HGVSC = 'TFX_HGVSC'
+    TFX_HGVSP1 = 'TFX_HGVSP1'
+    TFX_HGVSP3 = 'TFX_HGVSP3'
+    TFX_SPLICE = 'TFX_SPLICE'
+    TFX_VARIANT_EFFECT = 'TFX_VARIANT_EFFECT'
+    TFX_VARIANT_TYPE = 'TFX_VARIANT_TYPE'
+    TFX_BASE_POSITION = 'TFX_BASE_POSITION'
+    TFX_PROTEIN_TRANSCRIPT = 'TFX_PROTEIN_TRANSCRIPT'
+    TFX_EXON = 'TFX_EXON'
+    TFX_AMINO_ACID_POSITION = 'TFX_AMINO_ACID_POSITION'
+     
+def _parse_args():
+    '''
+    Validate and return command line arguments.
+    '''
+    parser = argparse.ArgumentParser(description='Produce a VCF containing transcript details generated by tx_eff.hgvs.py')
+
+    parser.add_argument('-i', '--in_vcf', 
+                    help='Input VCF', 
+                    type=argparse.FileType('r'), 
+                    required=True)
+
+    parser.add_argument('-o', '--out_vcf', 
+                    help='Output VCF', 
+                    type=argparse.FileType('w'), 
+                    required=True)
+
+    parser.add_argument('-c', '--in_csv', 
+                    help='Input CSV', 
+                    type=argparse.FileType('r'), 
+                    required=True)
+    
+    parser.add_argument('--version', action='version', version='%(prog)s ' + VERSION)
+    
+    args = parser.parse_args()
+    
+    return args
+    
+    args = parser.parse_args()
+
+
+def _read_vcf(vcf_filename: str):
+    '''
+    Read each variant from the VCF file and return a map of Variant[vcf-record]. Each variant must be unique and have only one ALT allele.  
+    '''
+    vcf_reader = vcfpy.Reader.from_path(vcf_filename)
+    variant_dict = dict()
+    
+    for vcf_record in vcf_reader:
+        if len(vcf_record.ALT) > 1:
+            raise Exception(f'VCF variants must have just one ALT allel: {vcf_record.CHROM}-{vcf_record.POS}-{vcf_record.REF}-{vcf_record.ALT}')
+        
+        variant = Variant(vcf_record.CHROM, vcf_record.POS, vcf_record.REF, vcf_record.ALT[0].value)
+
+        if variant_dict.get(variant):
```

Comment:
I'd only use get()if you intend to actually use the dict value if it does exist.  The "in" keyword is more pythonic I think, though these are functionally equivalent.